Explorer/Assets/DCL/UI/ConnectionStatusPanel/ConnectionStatusPanelController.cs

```diff
@@ -0,0 +1,146 @@
+using Arch.Core;
+using Cysharp.Threading.Tasks;
+using DCL.Multiplayer.Connections.Rooms.Status;
+using DCL.UI.ConnectionStatusPanel.StatusEntry;
+using DCL.UI.ErrorPopup;
+using DCL.UserInAppInitializationFlow;
+using DCL.Utilities;
+using ECS.SceneLifeCycle;
+using ECS.SceneLifeCycle.CurrentScene;
+using LiveKit.Proto;
+using MVC;
+using System;
+using System.Threading;
+
+namespace DCL.UI.ConnectionStatusPanel
+{
+    public partial class ConnectionStatusPanelController : ControllerBase<ConnectionStatusPanelView>
+    {
+        private readonly IUserInAppInitializationFlow userInAppInitializationFlow;
+        private readonly IMVCManager mvcManager;
+        private readonly ICurrentSceneInfo currentSceneInfo;
+        private readonly ECSReloadScene ecsReloadScene;
+        private readonly IRoomsStatus roomsStatus;
+        private readonly World world;
+        private readonly Entity playerEntity;
+        private readonly CancellationTokenSource cancellationTokenSource = new ();
+        private bool isSceneReloading;
+
+        public override CanvasOrdering.SortingLayer Layer => CanvasOrdering.SortingLayer.Persistent;
+
+        public ConnectionStatusPanelController(
+            ViewFactoryMethod viewFactory,
+            IUserInAppInitializationFlow userInAppInitializationFlow,
+            IMVCManager mvcManager,
+            ICurrentSceneInfo currentSceneInfo,
+            ECSReloadScene ecsReloadScene,
+            IRoomsStatus roomsStatus,
+            World world,
+            Entity playerEntity
+        ) : base(viewFactory)
+        {
+            this.userInAppInitializationFlow = userInAppInitializationFlow;
+            this.mvcManager = mvcManager;
+            this.currentSceneInfo = currentSceneInfo;
+            this.ecsReloadScene = ecsReloadScene;
+            this.roomsStatus = roomsStatus;
+            this.world = world;
+            this.playerEntity = playerEntity;
+        }
+
+        protected override void OnViewInstantiated()
+        {
+            currentSceneInfo.SceneStatus.OnUpdate += SceneStatusOnUpdate;
+            SceneStatusOnUpdate(currentSceneInfo.SceneStatus.Value);
+            Bind(roomsStatus.ConnectionQualityScene, viewInstance.SceneRoom);
+            Bind(roomsStatus.ConnectionQualityIsland, viewInstance.GlobalRoom);
+        }
+
+        private void SceneStatusOnUpdate(ICurrentSceneInfo.Status? obj)
+        {
+            const float DELAY = 5f;
+
+            async UniTaskVoid ShowButtonAsync()
+            {
+                await UniTask.Delay(TimeSpan.FromSeconds(DELAY));
+                viewInstance.Scene.ShowReloadButton(TryReloadScene);
+            }
+
+            if (obj is not { } status)
```

Comment:
Not clear what this line is doing, can you refactor? Just simplify to `obj != null`

Reply:
It basically says: if it is not an object otherwise it's "status" by one line

Reply:
Yeah, but this kind of syntax makes sense when you can validate a property, like: `if (obj is { Enabled })`, but expressing just `{ }` pollutes the reading. Its more clear if declared as: `if (obj!= null)`